pkg/agent/agent.go

```diff
@@ -448,12 +480,20 @@ func (i *Initializer) initNodeLocalConfig() error {
 		klog.Errorf("Failed to parse subnet from CIDR string %s: %v", node.Spec.PodCIDR, err)
 		return err
 	}
+	ip, err := noderoute.GetNodeAddr(node)
+	if err != nil {
+		klog.Errorf("Failed to obtain local IP address from k8s")
+		return err
+	}
+	localAddr, dev, err := util.GetDefaultLocalNodeAddr(ip)
+	if err != nil {
+		return fmt.Errorf("initNodeLocalConfig %v", err)
+	}
 
-	i.nodeConfig = &types.NodeConfig{Name: nodeName, PodCIDR: localSubnet}
+	i.nodeConfig = &types.NodeConfig{Name: nodeName, PodCIDR: localSubnet, NodeIPAddr: localAddr, NodeDefaultDev: dev.Attrs().Name}
```

Comment:
Why not just use the IP on K8s Node? Anyway for remote Nodes we know only their IPs from K8s.

Reply:
IP from k8s node is single IP, whereas NodeIPAddr is IP/Mask, needed to subnet testing with peer node

Reply:
I am wrong here. I got you need to know the subnet.